packages/backend/src/build-disk-image.ts

```diff
@@ -150,51 +122,85 @@ export async function buildDiskImage(imageData: unknown, history: History) {
         // Do progress report here so it doesn't look like it's stuck
         // since we are going to pull an image
         progress.report({ increment: 4 });
-        await containerUtils.pullImage(buildImageContainer.Image);
-
-        // delete previous copies of the image (in case we have upgraded it)
-        await containerUtils.deleteOldImages(image.engineId, buildImageContainer.Image);
+        if (buildImageContainer.Image) {
+          await containerUtils.pullImage(buildImageContainer.Image);
+        } else {
+          throw new Error('No image to pull');
+        }
 
         // Step 2. Check if there are any previous builds and remove them
         progress.report({ increment: 5 });
-        await containerUtils.removeContainerIfExists(image.engineId, buildImageContainer.name);
+        if (buildImageContainer.name) {
+          await containerUtils.removeContainerIfExists(build.engineId, buildImageContainer.name);
+        } else {
+          throw new Error('No container name to remove');
+        }
 
         // Step 3. Create and start the container for the actual build
         progress.report({ increment: 6 });
-        const containerId = await containerUtils.createAndStartContainer(image.engineId, buildImageContainer);
+        build.status = 'running';
+        await history.addOrUpdateBuildInfo(build);
+        const containerId = await containerUtils.createAndStartContainer(build.engineId, buildImageContainer);
+
+        // Update the history with the container id that was used to build the image
+        build.buildContainerId = containerId;
+        await history.addOrUpdateBuildInfo(build);
 
         // Step 3.1 Since we have started the container, we can now go get the logs
-        await logContainer(image, containerId, progress, data => {
+        await logContainer(build.engineId, containerId, progress, data => {
           logData += data;
         });
 
         // Step 4. Wait for the container to exit
         // This function will ensure it exits with a zero exit code
         // if it does not, it will error out.
         progress.report({ increment: 7 });
-        await containerUtils.waitForContainerToExit(containerId);
+
+        try {
+          await containerUtils.waitForContainerToExit(containerId);
+        } catch (error) {
+          // If we error out, BUT the container does not exist in the history, we will silently error
+          // as it's possible that the container was removed by the user during the build cycle / deleted from history.
+
+          // Check if history.infos has an entry with a containerId
+          const historyExists = history.infos.some(info => info.buildContainerId === containerId);
+          if (!historyExists) {
+            console.error(
+              'Container has errored out, but there is no history entry for it. This is likely due to the container being removed intentionally during the build cycle. Ignore this. Error:',
```

Comment:
IMHO from end user, you can't really say 'for it'
I would expect to have message about for example the image being built or the container name, etc